packages/profiling-node/src/integration.ts

```diff
@@ -172,15 +172,47 @@ class ContinuousProfiler {
    */
   public initialize(client: NodeClient): void {
     this._client = client;
+
+    // There is no off method to disable this, so we need to ensure to add the listener only once. This adds overhead
+    // to the event processing, but it is minimal as we short circuit if there is no profilerId active and return early.
+    this._client.on('beforeSendEvent', this._onBeforeSendThreadContextAssignment.bind(this));
```

Comment:
`on` should return a function that deregisters a callback. We can use that to disable it.

Reply:
Ahh, I missed that! I'll keep the same implementation I think bc of simplicity, but let me remove that comment

Reply:
Removed it and left a comment for future optimization